### DISCUSSION TITLE
Cargo error: "could not compile 'async-stripe' (lib)

### DISCUSSION THREAD

**Discussion by djmcmath** [arlyon/async-stripe#458] [Q&A] [answered]
Hello -- forgive the noob-question, but I'm having a hard time making async-stripe compile.  I suspect that I've included it incorrectly in my Cargo.toml, but that's just a guess.  The error is consistently "could not compile 'async-stripe' (lib)". (Full error at bottom, in case that's helpful)

I'm running rustc 1.74.1, freshly updated on an Ubuntu VM.

The relevant lines from my Cargo.toml are:
```
async-std = "^1.8, <1.11"
async-stripe = { version = "0.26.0", default-features=false, features = ["runtime-async-std-surf", "checkout"]}
```

Things I've Also Tried:
- tokio, with runtime-tokio-hyper and runtime-tokio-hyper-rustls
- different versions of async-stripe, down to 0.12
- Using * for the version
- No features at all, just default features off and the runtime

None of it works.

I'm not sure which version to reference here -- crates.io says to use 0.26.0.  But the async-stripe docs say to use the version that corresponds to the Stripe API that you're using.  From the readme:

```
The latest supported version of the Stripe API is 2020-08-27. Set the corresponding crate version depending on which version of the Stripe API you are pinned to. If you don't see the specific version you are on, prefer the next available version.

0.14 - stripe version 2020-08-27
0.12 - stripe version 2019-09-09
```

But my Stripe Dashboard says I'm using the 2023-10-16 API, which ... isn't either of the ones that are listed here.  But it doesn't seem to matter -- whatever number I pick, it doesn't compile.

So I guess:
- I've tried a bunch of things.
- I've read the docs.
- I've searched and can't find anything that looks right here or on SO.
- This thing is getting a ton of downloads, which means it must work great, so clearly this is on me.
- There's something going on here that I don't understand, and it's bad enough I don't even know where to look next.

I appreciate any help I can get on this.  

Thanks,
Dan

```
Caused by:
  process didn't exit successfully: `/home/djmcmath/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/bin/rustc --crate-name stripe --edition=2021 /home/djmcmath/.cargo/registry/src/index.crates.io-6f17d22bba15001f/async-stripe-0.26.0/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --diagnostic-width=263 --crate-type lib --emit=dep-info,metadata,link -C embed-bitcode=no -C debuginfo=2 --cfg 'feature="async"' --cfg 'feature="billing"' --cfg 'feature="checkout"' --cfg 'feature="chrono"' --cfg 'feature="connect"' --cfg 'feature="default"' --cfg 'feature="events"' --cfg 'feature="fraud"' --cfg 'feature="full"' --cfg 'feature="futures-util"' --cfg 'feature="hex"' --cfg 'feature="hmac"' --cfg 'feature="hyper"' --cfg 'feature="hyper-client"' --cfg 'feature="hyper-rustls"' --cfg 'feature="hyper-rustls-native"' --cfg 'feature="issuing"' --cfg 'feature="orders"' --cfg 'feature="runtime-tokio-hyper-rustls"' --cfg 'feature="sha2"' --cfg 'feature="sigma"' --cfg 'feature="stream"' --cfg 'feature="tax-calculation"' --cfg 'feature="terminal"' --cfg 'feature="tokio"' --cfg 'feature="uuid"' --cfg 'feature="webhook-endpoints"' --cfg 'feature="webhook-events"' -C metadata=deb9dc5fcdc2d819 -C extra-filename=-deb9dc5fcdc2d819 --out-dir /home/djmcmath/webserver/backend/target/x86_64-unknown-linux-gnu/debug/deps --target x86_64-unknown-linux-gnu -L dependency=/home/djmcmath/webserver/backend/target/x86_64-unknown-linux-gnu/debug/deps -L dependency=/home/djmcmath/webserver/backend/target/debug/deps --extern chrono=/home/djmcmath/webserver/backend/target/x86_64-unknown-linux-gnu/debug/deps/libchrono-0e0490f1aec2a255.rmeta --extern futures_util=/home/djmcmath/webserver/backend/target/x86_64-unknown-linux-gnu/debug/deps/libfutures_util-bd55676e795884c3.rmeta --extern hex=/home/djmcmath/webserver/backend/target/x86_64-unknown-linux-gnu/debug/deps/libhex-ab209f0f3c2b51c8.rmeta --extern hmac=/home/djmcmath/webserver/backend/target/x86_64-unknown-linux-gnu/debug/deps/libhmac-22824bde13be5ee7.rmeta --extern http_types=/home/djmcmath/webserver/backend/target/x86_64-unknown-linux-gnu/debug/deps/libhttp_types-02d1db93617a9fb7.rmeta --extern hyper=/home/djmcmath/webserver/backend/target/x86_64-unknown-linux-gnu/debug/deps/libhyper-b1b4d0a4aa4ad94c.rmeta --extern hyper_rustls=/home/djmcmath/webserver/backend/target/x86_64-unknown-linux-gnu/debug/deps/libhyper_rustls-0ab159932678b246.rmeta --extern serde=/home/djmcmath/webserver/backend/target/x86_64-unknown-linux-gnu/debug/deps/libserde-61a851e57370d6dd.rmeta --extern serde_json=/home/djmcmath/webserver/backend/target/x86_64-unknown-linux-gnu/debug/deps/libserde_json-a8f6c4b4cae6e73a.rmeta --extern serde_path_to_error=/home/djmcmath/webserver/backend/target/x86_64-unknown-linux-gnu/debug/deps/libserde_path_to_error-9e6ae71d038c0a08.rmeta --extern serde_qs=/home/djmcmath/webserver/backend/target/x86_64-unknown-linux-gnu/debug/deps/libserde_qs-c08186c65d0dff93.rmeta --extern sha2=/home/djmcmath/webserver/backend/target/x86_64-unknown-linux-gnu/debug/deps/libsha2-0467a4354674abb9.rmeta --extern smart_default=/home/djmcmath/webserver/backend/target/debug/deps/libsmart_default-22bf7da5e8e0fa2f.so --extern smol_str=/home/djmcmath/webserver/backend/target/x86_64-unknown-linux-gnu/debug/deps/libsmol_str-6411cf5497e86f5a.rmeta --extern thiserror=/home/djmcmath/webserver/backend/target/x86_64-unknown-linux-gnu/debug/deps/libthiserror-657328ac84b47ee6.rmeta --extern tokio=/home/djmcmath/webserver/backend/target/x86_64-unknown-linux-gnu/debug/deps/libtokio-f88cde55455e2bf0.rmeta --extern uuid=/home/djmcmath/webserver/backend/target/x86_64-unknown-linux-gnu/debug/deps/libuuid-c514f7abdb1189f4.rmeta --cap-lints allow -L native=/home/djmcmath/webserver/backend/target/x86_64-unknown-linux-gnu/debug/build/ring-9d65be97da2e9e79/out` (signal: 9, SIGKILL: kill)
```

**Comment by arlyon** [comment]
Yeah sorry the readme is outdated. You will probably need the 'core' feature as well as it contains a number of APIs that all the other chunks rely on.

**Reply by djmcmath** [reply]
Ok, so that's helpful.  I don't see how to add "core," however.  In the code, I find core as a module within the path "resources," but I have no idea how to add that.

What are the lines from Cargo.toml that are actually required to make this work?

**Reply by arlyon** [reply]
The easiest solution is to not disable default features. We are working on the `next` branch which gets rid of feature flags altogether and significantly improves the compile time also which will be a valid option soon-ish. I will update here when that is the case.

**Reply by djmcmath** [reply]
Would it be easier to update the readme than to push out the next branch?  I'm just thinking there's probably a short term path that'd help entry-level users of the library (read: people like me who can barely spell "rust") get it right, while you're working on the better long term solution.

I've found a workaround (pipe all of my Stripe interactions through a node.js server), but it's ugly.  I guess I'll revisit this when you get the next version out.

Thanks for building this.

**Reply by arlyon** [reply, marked as answer]
Took some time to verify this and it tools like the version you are specifying builds as expected. Ignore the stuff about the `core` feature, that is no longer relevant. This is the full contents of the Cargo.toml from my test project using the most recent version.

```toml
[package]
name = "async-stripe-test"
version = "0.1.0"
edition = "2021"

[dependencies]
async-stripe = { version = "0.37.2", default-features = false, features = [
    "runtime-async-std-surf",
] }
```

Commands to get here are as follows

```sh
cargo new async-stripe-test
cd async-stripe-test
cargo add async-stripe --no-default-features -F runtime-async-std-surf
```

**Comment by miyukki** [comment]
I got the same error when building with Docker. I increased the memory allocation from 8GB and it worked fine.
